### DISCUSSION TITLE
Sample of the clean block in PS 7.3

### DISCUSSION THREAD

**Discussion by ronascentes** [PowerShell/PowerShell#19638] [Q&A] [answered]
The clean block was added in PowerShell 7.3 but I have not found an good example of this implementation in a script. So can anyone provide a real sample of code when I might use the clean block in PS 7.3?

**Comment by kasini3000** [comment, marked as answer]
Scriptblock  (instead of PS1 script), function, foreach objetc pipeline, added
Clean {This code block is placed after end to clear resources, close connections, etc.}

https://github.com/PowerShell/PowerShell-RFC/blob/master/Archive/Experimental/RFC0059-Cleanup-Script-Block.md

Exiting from the Clean block will terminate execution within the Clean block
But ExitException will not propagate from the Clean block
Ctrl+c will not terminate clean
Note: There are no begin, process, or end in the script scope ( xxx.ps1 ).

**Comment by ronascentes** [comment]
Thank you @kasini3000
